src/index.ts

```diff
@@ -56,24 +66,59 @@ function getRecordExtractorSource(): string {
 }`;
 }
 
-function addComment(crawlerId: string): void {
-  try {
-    const pathArray = CRAWLER_API_BASE_URL.split('/');
-    const protocol = pathArray[0];
-    const host = pathArray[2];
-    const baseUrl = `${protocol}//${host}`;
+function findCommentPredicate(crawlerId: string, comment: Comment): boolean {
+  return comment.body ? comment.body.includes(crawlerId) : true;
+}
 
-    const message = `<p>Check your created <a href="${baseUrl}/admin/crawlers/${crawlerId}/overview" target="_blank">Crawler</a></p>
-    <p>Check your created index on your <a href="https://www.algolia.com/apps/${ALGOLIA_APP_ID}/explorer/browse/${CRAWLER_NAME}" target="_blank">Algolia Application</a></p>`;
+async function findComment(
+  prNumber: number,
+  crawlerId: string
+): Promise<Comment | undefined> {
+  const parameters = {
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: prNumber,
+  };
 
+  for await (const { data: comments } of octokit.paginate.iterator(
+    octokit.rest.issues.listComments,
+    parameters
+  )) {
+    // Search each page for the comment
+    const gaComment = comments.find((comment) =>
+      findCommentPredicate(crawlerId, comment)
+    );
+    if (gaComment) return gaComment;
+  }
+
+  return undefined;
+}
+
+async function addComment(crawlerId: string): Promise<void> {
+  try {
     const context = github.context;
     if (context.payload.pull_request === undefined) {
       core.info('No pull request found.');
       return;
     }
     const prNumber = context.payload.pull_request.number;
 
-    const octokit = github.getOctokit(GITHUB_TOKEN);
+    // First check if the comment doesn't already exist
+    const comment = await findComment(prNumber, crawlerId);
```

Comment:
Not sure what is the best DX. This one is perfectly valid. 
We could also delete the previous and post a new one. 
Or edit the old one just to be put the last update date for transparency.

Wdyt ?

Reply:
I decided just to ignore it in case it's already existing because the comment would be exactly the same, but I guess I can try to edit it.

Reply:
<img width="668" alt="image" src="https://user-images.githubusercontent.com/1649459/125804614-aba9a8a9-3f34-46c1-9623-d014c2b4ff33.png">

The comment should have been updated even though it's not written "edited" on the UI (maybe it's because the body is the same ?)

Reply:
Adding a date would help then or maybe the commit hash like Netlify